packages/graphql/tests/integration/directives/cypher/filtering/cypher-filtering-aggregation.test.ts

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TestHelper } from "../../../../utils/tests-helper";
+
+describe("cypher directive filtering - Aggregation", () => {
+    const testHelper = new TestHelper();
+    const Movie = testHelper.createUniqueType("Movie");
+
+    afterEach(async () => {
+        await testHelper.close();
+    });
+
+    test.each([
+        {
+            title: "String aggregation - shortest, filter on String",
+            customCypherType: "String",
+            createCypher: `
+                CREATE (:${Movie} { title: "The Matrix", released: 1999, custom_field: "hello" })
+                CREATE (:${Movie} { title: "The Matrix Reloaded", released: 2003, custom_field: "hello world" })
```

Comment:
Would be cool to have something like the following to really make sure this stuff is working:

```suggestion
                CREATE (:${Movie} { title: "The Matrix", released: 1999, custom_field: "hello" })
                CREATE (:${Movie} { title: "The Matrix Reloaded", released: 2003, custom_field: "hello world" })
                CREATE (:${Movie} { title: "The", released: 2003, custom_field: "goodbye world" })
```

We should still expect "The Matrix" as the result, and trouble otherwise! Easy to rinse and repeat similar cases easily throughout this file.

Reply:
I've added what I think are additional data points to show the filtering working correctly.